client/smart_contract.go

```diff
@@ -0,0 +1,71 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/rubixchain/rubixgoplatform/core/model"
+)
+
+type SmartContractRequest struct {
+	BinaryCode string
+	RawCode    string
+	SchemaCode string
+	DID        string
+	SCPath     string
+}
+
+type FetchSmartContractRequest struct {
+	SmartContractToken     string
+	SmartContractTokenPath string
+}
+
+func (c *Client) GenerateSmartContractToken(smartContractRequest *SmartContractRequest) (*model.BasicResponse, error) {
+
+	fields := make(map[string]string)
+	files := make(map[string]string)
+
+	if smartContractRequest.BinaryCode != "" {
+		files["binaryCodePath"] = smartContractRequest.BinaryCode
+	}
+	if smartContractRequest.RawCode != "" {
+		files["rawCodePath"] = smartContractRequest.RawCode
+	}
+	if smartContractRequest.SchemaCode != "" {
+		files["schemaFilePath"] = smartContractRequest.SchemaCode
+	}
+	if smartContractRequest.DID != "" {
+		fields["did"] = smartContractRequest.DID
+	}
+
+	for key, value := range fields {
+		fmt.Printf("Field: %s, Value: %s\n", key, value)
+	}
+
+	for key, value := range files {
+		fmt.Printf("File: %s, Value: %s\n", key, value)
+	}
+
+	var basicResponse model.BasicResponse
+	err := c.sendMutiFormRequest("POST", "/api/generate-smart-contract", nil, fields, files, &basicResponse)
```

Comment:
We can assign the URL to generate and fetch to a constant value, as we have followed for other API URL also